lib/util/template/index.spec.ts

```diff
@@ -85,6 +85,40 @@ describe('util/template/index', () => {
     expect(output).toBe('foo');
   });
 
+  it('has access to basic environment variables (basicEnvVars)', () => {
+    const userTemplate = 'HOME is {{env.HOME}}';
+    const output = template.compile(userTemplate, {});
+    expect(output).toBe(`HOME is ${process.env.HOME ?? ''}`);
+  });
+
+  it('and has access to environment variables (can be exposed with customEnvVariables)', () => {
+    const input = {
+      env: {
+        SHELL: process.env.SHELL,
```

Comment:
@viceice Here's the problem I am facing:

- this file imports the module `template`;
- consts `allowedTemplateFields` and `allowedFieldsList` get initialized when the module is loaded;
- no matter what I do in the unit test - even if I added `SHELL` to `GlobalConfig.customEnvVariables` - `SHELL` will not be added to the allowed fields.

How should I construct this test?

I can see two options:

1. In Python, I would override the consts values set in the module by monkeypatching them for the test. `mockReturnValueOnce` seems promising.
2. Reloading `template` after setting `env` and `customEnvVariables` would solve the issue.

Which is the proper way to go?

Reply:
OK, I found a way. Not sure if it is the most elegant one, but the tests are :green_circle:.